features/core/src/main/java/org/apache/karaf/features/internal/service/Deployer.java

```diff
@@ -110,26 +110,12 @@
 public class Deployer {
 
     public interface DeployCallback {
-
         void print(String message, boolean verbose);
-
         void saveState(State state);
         void persistResolveRequest(DeploymentRequest request) throws IOException;
         void installFeature(Feature feature) throws IOException, InvalidSyntaxException;
```

Comment:
I think installFeature is very similar to uninstallFeature. So we would need to move both. If you look what they depend on there would not be much left in FeaturesServiceImpl. 

I have some ideas how to improve the situation but they would require a bigger redesign.